packages/vite/src/node/plugins/css.ts

```diff
@@ -664,38 +664,47 @@ async function compileCSS(
 interface PostCSSConfigResult {
   options: Postcss.ProcessOptions
   plugins: Postcss.Plugin[]
+  file?: string
 }
 
-let cachedPostcssConfig: PostCSSConfigResult | null | undefined
+let cachedPostcssConfigPath: string | undefined | null
 
 async function resolvePostcssConfig(
+  file: string,
   config: ResolvedConfig
 ): Promise<PostCSSConfigResult | null> {
-  if (cachedPostcssConfig !== undefined) {
-    return cachedPostcssConfig
-  }
-
   // inline postcss config via vite config
   const inlineOptions = config.css?.postcss
+
   if (isObject(inlineOptions)) {
     const result = {
       options: { ...inlineOptions },
       plugins: inlineOptions.plugins || []
     }
     delete result.options.plugins
-    return (cachedPostcssConfig = result)
+
+    return result
+  }
+
+  if (cachedPostcssConfigPath === null) {
+    return null
   }
 
   try {
     const searchPath =
-      typeof inlineOptions === 'string' ? inlineOptions : config.root
-    // @ts-ignore
-    return (cachedPostcssConfig = await postcssrc({}, searchPath))
+      cachedPostcssConfigPath ??
+      (typeof inlineOptions === 'string' ? inlineOptions : config.root)
+
+    const ctx = { file, env: config.mode }
+    const result = (await postcssrc(ctx, searchPath)) as PostCSSConfigResult
+    cachedPostcssConfigPath = result.file
+
+    return result
   } catch (e) {
     if (!/No PostCSS Config found/.test(e.message)) {
       throw e
     }
-    return (cachedPostcssConfig = null)
+    return (cachedPostcssConfigPath = null)
```

Comment:
**thought (dx):** I'm a bit confused by this line
Could we split it into two lines?
I think it means the same as :thinking:  

```ts
cachedPostcssConfigPath = null
return null
```